src/view-canvas/app-view.ts

```diff
@@ -0,0 +1,102 @@
+import { Coords } from "../utility/coords";
+import { App } from "../app-states/app";
+import { Vector } from "../utility/vector";
+import { AnimationFrame } from "../utility/animation";
+
+export class Sprite {
+    constructor(private texture: CanvasImageSource) {}
+
+    draw(context: CanvasRenderingContext2D, options: {
+        position: Vector,
+        frame: AnimationFrame,
+        rotation: number
+    }) {
+        const translation = new Vector(
+            options.position.x,// + options.frame.w / 2,
+            options.position.y// + options.frame.h / 2
+        );
+
+        context.save();
+        context.translate(
+            translation.x, 
+            translation.y);
+        context.rotate((options.rotation + 90) * Math.PI/180.0);
+        context.translate(
+            -translation.x,
+            -translation.y);
+        context.drawImage(
+            this.texture, 
+            options.frame.x,
+            options.frame.y,
+            options.frame.w, 
+            options.frame.h, 
+            options.position.x - options.frame.w / 2,
+            options.position.y - options.frame.h / 2,
+            options.frame.w, 
+            options.frame.h);
```

Comment:
This effectively hard-couples game objects and rendering together again :( Why should game-object know which sprite to draw and - most importantly - how big it is?

Reply:
As I explained IRL, the important bit is that game objects have access to animation engine, so they can change the animations and can be notified when animation ends. However, object that initializes the AnimationEngine can come from outside and not be tied to game logic.

But even if you do so, you still need to have some sense of scale between rendering and game logic. If the sprite is 16x16px, the collider can't have bigger radius than 8 (and the collider will likely be smaller) and then the render engine would have to know how to scale sprites if collider radius is XY.

So I esentially agree with your comment below which I just noticed is here, I just want to point out that on some level, your GameObject is dependent on sprite size (which is btw why so game frameworks tie Sprite and collider together, I saw something along these lines even in PIXI).